hazelcast-client/src/main/java/com/hazelcast/client/connection/nio/ClusterConnector.java

```diff
@@ -282,7 +282,9 @@ public void run() {
     }
 
     public void shutdown() {
-        ClientExecutionServiceImpl.shutdownExecutor("cluster", clusterConnectionExecutor, logger);
+        // since the connector is shutting down, we don't care about any remaining
+        // task. So we want to shutdown fast.
+        shutdownNowExecutor("cluster", clusterConnectionExecutor, logger);
```

Comment:
Did you see what this executor is waiting for? Did we understand the reason of the delay?

Reply:
The logic is changed so that pending tasks are not processed. So if they are not processed, there is no delay.